xline-client/src/lease_gen.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use clippy_utilities::Cast;
+
+/// Generator of unique lease id
+/// id format:
+/// | prefix       | suffix  |
+/// | 7 bytes      | 1 byte  |
+/// | random bytes | cnt     |
+#[derive(Debug, Default)]
+pub struct LeaseIdGenerator {
+    /// prefix of id
+    prefix: u64,
+    /// suffix of id
+    suffix: AtomicU64,
+}
+
+impl LeaseIdGenerator {
+    /// New `IdGenerator`
+    ///
+    /// # Panics
+    ///
+    /// panic if failed to generate random bytes
+    #[inline]
+    #[must_use]
+    pub fn new() -> Self {
+        let mut buf = [0u8; 8];
+        getrandom::getrandom(&mut buf).unwrap_or_else(|err| {
+            panic!("Failed to generate random bytes for lease id generator: {err}");
+        });
+        let prefix = u64::from_be_bytes(buf).overflowing_shl(1).0;
```

Comment:
Here should be `overflowing_shl(8)` in that a byte equals eight bits. FYI: https://doc.rust-lang.org/std/primitive.u64.html#method.overflowing_shl